### ISSUE TITLE
Any plans on creating a xmlrpc server?

### ISSUE THREAD

**Issue by TimWhiting** [a14n/dart-xmlrpc#6]
I'm interested in xmlrpc server api. This library looks very good, and I've verified that it works for me, but it's missing the server implementation. Is that something you have planned or would accept a pull request for?

**Comment by Elblassy** [comment]
0.2.4 version works with you ? 
it get me this error : Because every version of flutter_test from sdk depends on meta 1.1.6 and xml_rpc >=0.2.3 depends on meta ^1.1.8, flutter_test from sdk is incompatible with xml_rpc >=0.2.3.

**Comment by a14n** [comment]
@TimWhiting you're correct. There is no server implementation for now.

PRs are always welcome but I'm not sure where the server support should end. Perhaps we should only provide a way to encode response and fault and let other package/code deal with http handling.

**Comment by TimWhiting** [comment]
@a14n 

I'm currently working on a server implementation in my fork. I've got the basics working, just need to add in fault handling. I've structured it similarly to the python implementation, but am open to other suggestions. I'll be adding in fault handling next, and adding tests, documentation comments etc.

For the http server part, I'm just using the HttpServer class from dart:io and sending back a 404 error for anything except a POST request that is in the XMLRPC format. I didn't see a need to use an external library since currently the whole thing is only about 200 additional lines of code. But for more complex use cases than my own where you might want a multi-threaded server or something, it might be good to use something else. The organization should be modular and enable both use cases.

Probably later in the week, once I add in some more documentation and examples, I'll send a pull request, unless you want to follow the progress or guide the implementation more.
